### PR TITLE
AZP: skip commit style check for non-PRs

### LINKED ISSUE #4538: AZP CI fails when running on direct pushes
### Describe the bug
AZP CI is triggered by both PRs and direct pushes to the master and release branches. A "Codestyle commit title" check expects `System.PullRequest.TargetBranch` to be set while it is not.

### Steps to Reproduce
Trigger AZP CI in a branch manually or push commits directly so it will be triggered automatically. Example: https://dev.azure.com/ucfconsort/ucx/_build/results?buildId=2930.

### PR DESCRIPTION
**Pull request by amaslenn**
## What
Disable commit title style check for direct pushes and merges. The check will look like [this](https://dev.azure.com/ucfconsort/ucx%20-%20test/_build/results?buildId=3147&view=logs&j=a3e7afe8-b282-5869-712a-9857fbd4b4a9&t=fd231581-c084-51fb-f10e-187ae5f1c153).
Fixes #4538 

## Why ?
Commit range is not available in this case.

### CODE REVIEW

File: buildlib/azure-pipelines.yml [hunk 1]
```diff
@@ -44,6 +44,7 @@ stages:
                  echo "##vso[task.logissue type=error]Bad commit title(s), see $url for more info."
                  echo "##vso[task.complete result=Failed;]"
               fi
+            condition: eq(variables['Build.Reason'], 'PullRequest')
```
Comment (yosefe) [inline buildlib/azure-pipelines.yml]: WDYT about checking System.PullRequest.TargetBranch is non-empty?

Comment (amaslenn) [reply buildlib/azure-pipelines.yml]: `System.PullRequest` is not defined at all for non-PRs. IMO checking the `Build.Reason` is a reliable way to check what we need. Do you see any potential issues with that?

Comment (yosefe) [reply buildlib/azure-pipelines.yml]: ok

**Comment by mellanox-github** [comment]
<details><summary>Mellanox CI: <b>PASSED</b#  > on 25 workers (click for details)</summary>

**Note:** the logs will be deleted after 23-Dec-2019

Agent/Stage | Status
-- | --
[_main](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/a1a12e2cdc22a147752da02a28320de1a630e2dc/_main) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/c707a6be2616b279cb66b7c3b847f50eb51b2d08/hpc-arm-cavium-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/74e85e9f598c65524230a99a023111a4b337e153/hpc-arm-cavium-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/77dfe5c7096de7e5520ed5a1a0c29713de14031c/hpc-arm-cavium-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-cavium-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/84fa076eb7f1a4f62de85dd4e0b3f64c6f8ab1f3/hpc-arm-cavium-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/5a62114907a6a2cb9d234669fb6fbdf612383f57/hpc-arm-hwi-jenkins_W0) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/c39f37026afb4e8d99d4734a851e338faa1c19ba/hpc-arm-hwi-jenkins_W1) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/48cf039b5cc4903778dba17f98265b44d6340686/hpc-arm-hwi-jenkins_W2) | :heavy_check_mark: SUCCESS
[hpc-arm-hwi-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/ceb84d249eee243a3f0b2b3a4584124628daa21d/hpc-arm-hwi-jenkins_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W0](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/27730eea3c8c7fe3f19b5c801cd88ed07f6edb11/hpc-test-node-gpu_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W1](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/cc770579d1f8b87983f6273abb8326ae2f712400/hpc-test-node-gpu_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W2](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/522b12d8d7b4687066e615a67976f017ff24cf34/hpc-test-node-gpu_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-gpu_W3](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/ca02096db7f52cd0446758aa4953f433221dafec/hpc-test-node-gpu_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W0](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/638b9cb6efef41a1586632855a731a58cee75a3b/hpc-test-node-legacy_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W1](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/3df752efd2ecfe2e791f1d16feb4e5eb2c7c23bc/hpc-test-node-legacy_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W2](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/4d8522dc0b8c6e23745c2c754f63a67983a801d8/hpc-test-node-legacy_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-legacy_W3](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/e4968fde59345b3af674099e01763e0e00d65096/hpc-test-node-legacy_W3) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W0](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/a203e2ff6cea689cdc34bbe817c2231ba959053d/hpc-test-node-new_W0) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W1](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/9b992dd6c8cb2f4dff26902a2bcc1735a484a661/hpc-test-node-new_W1) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W2](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/9f7a9b0bd6643aa2e3c4060c276fc2ed942fe648/hpc-test-node-new_W2) | :heavy_check_mark: SUCCESS
[hpc-test-node-new_W3](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/b62ad7badeb5ef98ed35dc25e9b0fe631d841a56/hpc-test-node-new_W3) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W0](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/e5591b0a1db77c25b82c32d65c557119620a35a6/r-vmb-ppc-jenkins_W0) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W1](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/97ff9a293edcc10a72588c17b428ebf9c58fe901/r-vmb-ppc-jenkins_W1) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W2](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/35ea815793957b6e022f1b34d289ab1962bfaec3/r-vmb-ppc-jenkins_W2) | :heavy_check_mark: SUCCESS
[r-vmb-ppc-jenkins_W3](https://gist.githubusercontent.com/mellanox-github/ad0016514f983afbd73fe104b5f9c76f/raw/9fd5db5a945c1100c62aed744e76b208d4fccaf7/r-vmb-ppc-jenkins_W3) | :heavy_check_mark: SUCCESS

**Pull request merged**